pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -678,12 +678,22 @@ def build_package(
             f"directory at the path {srcpath}, but that path does not exist."
         )
 
+    import os
+    import subprocess
+    import sys
+
+    tee = subprocess.Popen(["tee", pkg_root / "build.log"], stdin=subprocess.PIPE)
+    # Cause tee's stdin to get a copy of our stdin/stdout (as well as that
+    # of any child processes we spawn)
+    os.dup2(tee.stdin.fileno(), sys.stdout.fileno())  # type: ignore[union-attr]
+    os.dup2(tee.stdin.fileno(), sys.stderr.fileno())  # type: ignore[union-attr]
+
     with chdir(pkg_root), get_bash_runner() as bash_runner:
         bash_runner.env["PKG_VERSION"] = version
         if not continue_:
             prepare_source(pkg_root, build_dir, srcpath, source_metadata)
 
-        run_script(build_dir, srcpath, build_metadata, bash_runner)
+            run_script(build_dir, srcpath, build_metadata, bash_runner)
```

Comment:
Shouldn't we run `script` even if the build continues from the middle? We often set environment variables that are needed for the build.

Reply:
Well it depends, in scipy there is a command that concatenates a bunch of stuff into a file which we only want to do once. We should go back to being able to choose whether to run the script or not which used to be an option before #2256.